frontend/packages/process-editor/src/components/ConfigPanel/ConfigContent/EditDataTypes/SelectDataTypes/SelectDataTypes.tsx

```diff
@@ -44,26 +43,36 @@ export const SelectDataTypes = ({
 
   return (
     <div className={classes.dataTypeSelectAndButtons}>
-      <Combobox
+      <StudioCombobox
         label={t('process_editor.configuration_panel_set_data_model_label')}
-        value={currentValue && dataModelIds.includes(existingDataType) ? currentValue : undefined}
-        description={t('process_editor.configuration_panel_data_model_selection_description')}
+        value={
+          existingDataType && dataModelOptionsToDisplay.includes(existingDataType)
+            ? currentValue
+            : undefined
+        }
+        description={
+          existingDataType
+            ? t(
+                'process_editor.configuration_panel_data_model_selection_description_existing_model',
+              )
+            : t('process_editor.configuration_panel_data_model_selection_description')
+        }
```

Comment:
No biggie, but we can consider moving the logic for the value and description out of the combobox so it is easier to read 😊

Reply:
I agree! Moved the logic for value and description into separate variables 😊